localstack/services/apigateway/router_asf.py

```diff
@@ -31,7 +31,13 @@ def to_invocation_context(
         url_params = {}
 
     method = request.method
-    path = request.full_path if request.query_string else request.path
+    url_params_stage = url_params.get("stage", "dev")
+    url_params_path = url_params.get("path", request.path)
+    # Base path is not URL-decoded. Example: test%2Balias@gmail.com => test%2Balias@gmail.com
+    path = f"{url_params_stage}/{url_params_path}"
```

Comment:
Does the current solution work with custom domain names where the stage_name is omitted?